### PR TITLE
Package Signing for RPM and DEB

### PR DESCRIPTION
**Pull request by northrup**

### CODE REVIEW

File: Builds/containers/gitlab-ci/pkgbuild.yml [hunk 1]
```diff
@@ -131,6 +132,65 @@ dpkg_build:
...
+    fi
+  artifacts:
+    paths:
+      - build/rpm/packages/
+  cache:
+    key: rpm
+    paths:
+      - .nih_c/pkgbuild
+  script:
+    - . ./Builds/containers/gitlab-ci/sign_package.sh rpm
```
Comment (MarkusTeufelberger) [inline Builds/containers/gitlab-ci/pkgbuild.yml]: This script doesn't exist: https://github.com/ripple/rippled/tree/develop/Builds/containers/gitlab-ci

File: Builds/containers/gitlab-ci/pkgbuild.yml [hunk 2]
```diff
@@ -131,6 +132,65 @@ dpkg_build:
...
+##  stage: sign_packages                                               ##
+##                                                                     ##
+##  build packages using containers from previous stage.               ##
+##                                                                     ##
+#########################################################################
+
+rpm_sign:
+  stage: sign_packages
+  dependencies:
+    - build_centos_container
```
Comment (mellery451) [inline Builds/containers/gitlab-ci/pkgbuild.yml]: if we want to sign the packages that were built in the previous step, seems like these should depend on `rpm_build/dpkg_build` instead.

File: Builds/containers/gitlab-ci/pkgbuild.yml [hunk 3]
```diff
@@ -131,6 +132,65 @@ dpkg_build:
...
+      unset GPG_KEY_B64
+      export GPG_KEYID=$(gpg --with-colon --list-secret-keys | head -n1 | cut -d : -f 5)
+    else
+      echo -e "\033[0;31m****** GPG signing disabled ******\033[0m"
+      exit
+    fi
+  artifacts:
+    paths:
+      - build/rpm/packages/
+  cache:
```
Comment (mellery451) [inline Builds/containers/gitlab-ci/pkgbuild.yml]: can't say for sure without looking at the sign_package script, but I doubt this cache is useful for the signing (it only has build dependencies in it, but the package build should be complete by this step)

File: Builds/containers/gitlab-ci/pkgbuild.yml (hunk 1 above)
Comment (mellery451) [reply Builds/containers/gitlab-ci/pkgbuild.yml]: agree with @MarkusTeufelberger ...I think this script needs to be added to the commit

File: Builds/containers/gitlab-ci/pkgbuild.yml [hunk 4]
```diff
@@ -131,6 +132,65 @@ dpkg_build:
...
+  before_script:
+  - |
+    # checking GPG signing support
+    if [ -n "$GPG_KEY_B64" ]; then
+      echo "$GPG_KEY_B64" | gpg --batch --no-tty --allow-secret-key-import --import -
+      unset GPG_KEY_B64
+      export GPG_KEYID=$(gpg --with-colon --list-secret-keys | head -n1 | cut -d : -f 5)
+    else
+      echo -e "\033[0;31m****** GPG signing disabled ******\033[0m"
+      exit
```
Comment (mellery451) [inline Builds/containers/gitlab-ci/pkgbuild.yml]: should we fail this step in this case? (e.g. `exit -1`) ?

**Comment by MarkusTeufelberger** [comment]
Hey there @northrup, would you mind developing your branch in your own fork instead of the main repository? I have notifications enabled for this one and you're spamming my timeline a bit with your commits.

**Pull request closed without merging**